src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -43,8 +43,19 @@ public void TestEnvironmentProperty()
             environment.Remove("NewKey");
             Assert.Equal(CountItems + 1, environment.Count);
 
-            //Exception not thrown with invalid key
-            Assert.Throws<ArgumentException>(() => { environment.Add("NewKey2", "NewValue2"); });
+            // Exception thrown with invalid key
+            // The full .NET framework represents Environment with StringDictionary. On .NET Core we use
+            // use Dictionary<string, string>, because we need to have case sensitive behaviour on Unix.
+            // The full .NET Framework uses CaseSensitiveStringDictionary, but we can't have an
+            // equivilent as we don't have access to the internal state of StringDictionary in .NET Core.
+            if (PlatformDetection.IsFullFramework)
+            {
+                environment.Add("NewKey2", "NewValue2");
+            }
+            else
+            {
+                Assert.Throws<ArgumentException>(null, () => environment.Add("NewKey2", "NewValue2"));
```

Comment:
Wow, 600 hits in our code for `Assert.Throws<ArgumentException>(null,` ... something for another day. I am pretty confident it's not a prohibited breaking change.